op-e2e/actions/l2_sequencer.go

```diff
@@ -0,0 +1,94 @@
+package actions
+
+import (
+	"github.com/ethereum-optimism/optimism/op-node/eth"
+	"github.com/ethereum-optimism/optimism/op-node/rollup"
+	"github.com/ethereum-optimism/optimism/op-node/rollup/derive"
+	"github.com/ethereum-optimism/optimism/op-node/rollup/driver"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/stretchr/testify/require"
+)
+
+// L2Sequencer is an actor that functions like a rollup node,
+// without the full P2P/API/Node stack, but just the derivation state, and simplified driver with sequencing ability.
+type L2Sequencer struct {
+	L2Verifier
+
+	sequencer        *driver.Sequencer
+	l1OriginSelector *driver.L1OriginSelector
+
+	seqOldOrigin bool // stay on current L1 origin when sequencing a block, unless forced to adopt the next origin
+
+	failL2GossipUnsafeBlock error // mock error
+}
+
+func NewL2Sequencer(log log.Logger, l1 derive.L1Fetcher, eng derive.Engine, cfg *rollup.Config, seqConfDepth uint64) *L2Sequencer {
+	ver := NewL2Verifier(log, l1, eng, cfg)
+	return &L2Sequencer{
+		L2Verifier:              *ver,
+		sequencer:               driver.NewSequencer(log, cfg, l1, eng),
+		l1OriginSelector:        driver.NewL1OriginSelector(log, cfg, l1, seqConfDepth),
```

Comment:
why is the conf depth not on the config? seems like it would be a sane place for it to live

Reply:
It's the a sequencer specific config. The conf depth is not a property of the rollup, but a property of a sequencer at a point in time (it's valid for the sequencer to change it's conf depth).